crates/engine/src/ext.rs

```diff
@@ -358,12 +410,39 @@ impl Engine {
         unimplemented!("off-chain environment does not yet support `call`");
     }
 
-    pub fn weight_to_fee(&self, _gas: u64, _output: &mut &mut [u8]) {
-        unimplemented!("off-chain environment does not yet support `weight_to_fee`");
+    /// Emulates gas price calculation.
+    pub fn weight_to_fee(&self, gas: u64, output: &mut &mut [u8]) {
+        let fee = self.chain_spec.gas_price.saturating_mul(gas.into());
+        let fee: Vec<u8> = scale::Encode::encode(&fee);
+        set_output(output, &fee[..])
     }
 
-    pub fn random(&self, _subject: &[u8], _output: &mut &mut [u8]) {
-        unimplemented!("off-chain environment does not yet support `random`");
+    /// Returns a randomized hash.
+    ///
+    /// # Note
+    ///
+    /// - This is the off-chain environment implementation of `random`.
+    ///   It provides the same behavior in that it will likely yield the
+    ///   same hash for the same subjects within the same block (or
+    ///   execution context).
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    ///    let engine = ink_engine::ext::Engine::default();
+    ///    let seed = [0u8; 32];
+    ///    let mut output = [0u8; 32];
+    ///    engine.random(&seed, &mut output.as_mut_slice());
```

Comment:
Let's make it consistent with the argument name.

```suggestion
    ///    let subject = [0u8; 32];
    ///    let mut output = [0u8; 32];
    ///    engine.random(&subject, &mut output.as_mut_slice());
```